service/history/handler/handler_test.go

```diff
@@ -3619,12 +3619,15 @@ func TestRatelimitUpdate(t *testing.T) {
 		},
 	})
 	require.NoError(t, err)
-	alg, err := algorithm.New(algorithm.Config{
-		NewDataWeight:  func(opts ...dynamicconfig.FilterOption) float64 { return 0.5 },
-		UpdateInterval: func(opts ...dynamicconfig.FilterOption) time.Duration { return 3 * time.Second },
-		DecayAfter:     func(opts ...dynamicconfig.FilterOption) time.Duration { return 6 * time.Second },
-		GcAfter:        func(opts ...dynamicconfig.FilterOption) time.Duration { return time.Minute },
-	})
+	alg, err := algorithm.New(
+		metrics.NewNoopMetricsClient(),
```

Comment:
nit: You can always verify your metrics with tally.NewTestScope() and actual metrics client.

Reply:
I don't believe that'd be useful in this test thing, but yea - a finer-grained test on the ratelimiter itself may be worth writing.  I think it'd be the first metrics-test that checks a histogram 🤔

Reply:
added over in requestweighted_test.go :)
and yea, it's our first histogram test - the helper in it might be useful elsewhere later, but for now it's just internal so it's a bit easier to deal with.